examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,412 @@
+#
+# 1. Using the file paths
+# 2. Using the in-memory objects
+#
+# So we will measure the memory usage for both cases.
+#
+# Let's first create a dictionary to store the memory usage for each method
+
+from memory_profiler import memory_usage
+
+nifti_masker = {"single": {"path": [], "in_memory": []}}
+
+nifti_masker["single"]["path"] = memory_usage(
+    (nifti_masker_single, (fmri_path, mask_paths[0])),
+    max_usage=True,
+)
+print(
+    "Peak memory usage: with NiftiMasker, single mask, with paths:\n"
+    f"{nifti_masker['single']['path']} MiB"
+)
+
+nifti_masker["single"]["in_memory"] = memory_usage(
+    (nifti_masker_single, (fmri_img, mask_imgs[0])),
+    max_usage=True,
+)
+print(
+    "Peak memory usage: with NiftiMasker, single mask, with in-memory image:\n"
+    f"{nifti_masker['single']['in_memory']} MiB"
+)
+
+# %%
+# Masking using NiftiMasker in parallel
+# -------------------------------------
+# Now let's see how we would mask the fMRI image using multiple masks in
+# parallel using the :mod:`joblib` package.
+#
+# Let's add another key to the previous dictionary to store the memory usage
+# for this case.
+
+from joblib import Parallel, delayed
+
+
+def nifti_masker_parallel(fmri_path, mask_paths):
+    return Parallel(n_jobs=N_REGIONS)(
+        delayed(nifti_masker_single)(fmri_path, mask) for mask in mask_paths
+    )
+
+
+nifti_masker["parallel"] = {"path": [], "in_memory": []}
+
+nifti_masker["parallel"]["path"] = memory_usage(
+    (nifti_masker_parallel, (fmri_path, mask_paths)),
+    max_usage=True,
+    include_children=True,
+    multiprocess=True,
+)
+print(
+    f"Peak memory usage: with NiftiMasker, {N_REGIONS} jobs in parallel, "
+    "with paths:\n"
+    f"{nifti_masker['parallel']['path']} MiB"
+)
+
+nifti_masker["parallel"]["in_memory"] = memory_usage(
+    (nifti_masker_parallel, (fmri_path, mask_paths)),
+    max_usage=True,
+    include_children=True,
+    multiprocess=True,
+)
+print(
+    f"Peak memory usage: with NiftiMasker, {N_REGIONS} jobs in parallel, "
+    "with in-memory images:\n"
+    f"{nifti_masker['parallel']['in_memory']} MiB"
+)
+
+
+# %%
+# Masking using numpy indexing
+# ----------------------------
+# Now let's see how we can mask the fMRI image using numpy indexing. This
+# could be more efficient than the NiftiMasker when we simply need to mask
+# the fMRI image with binary masks and don't need to standardize, smooth, etc.
+# the image.
+#
+# In addition, we will use :func:`nibabel.loadsave.load` function to load the
+# fMRI image as a proxy object. This will allow us to load the data directly
+# from the file as a numpy array without loading the entire image into memory.
+# You can find more information about this in the :mod:`nibabel` documentation,
+# here: https://nipy.org/nibabel/images_and_memory.html
+#
+# As before we will do this by loading the data from the file paths and from
+# the in-memory objects. This time we will define two functions to do that.
```

Comment:
I don't know that it makes sense to show the non-memmap example here ; I'm lightly against it to keep things clean

Reply:
> to keep things clean

I agree that the example does read a bit repetitive. But I would say the difference between path vs. in-memory is pretty significant and since this is an advanced example, it does fall within the scope.

Maybe removing the "single" instances would make it better?

Reply:
Yes, I think I'm leaning towards removing all of the `single` instances ; presumably folks can load a large array into memory at least once ! If not, then we're more likely in the "high resolution" rather than "long time series" case.